src/track/beatstats.cpp

```diff
@@ -0,0 +1,84 @@
+#include <QList>
+#include <QMap>
+
+#include <algorithm>
+#include <cmath>
+
+#include "track/beatstats.h"
+#include "util/fpclassify.h"
+
+
+void MovingAvarage::update(double newValue, double oldValue) {
+    if (!util_isnan(oldValue)) {
+        m_sum -= oldValue;
+    }
+    m_sum += newValue;
+}
+
+void MovingAvarage::reset() {
+    m_sum = 0;
+    clear();
+}
+
+double MovingAvarage::compute() {
+    return m_sum / static_cast<double>(windowSize());
+}
+
+void MovingMode::update(double newValue, double oldValue) {
+    m_tempoFrequency[newValue] += 1;
+    if (!util_isnan(oldValue)) {
```

Comment:
That's because of the way I implemented the window, when it's not full it returns a nan so we know not to remove any element. Makes sense? We are actually not testing if it`s nan but if the window is not full yet

Reply:
Ah Ok, Please comment that in the code. 
Please also secure it against input data being nan.